src/docker_hub.ml

```diff
@@ -22,18 +21,23 @@ let rec cmps = function
       | n -> n
 
 let hurl ~meth ~headers url =
-  Http_lwt_client.request
-    ~config:(`HTTP_1_1 Httpaf.Config.default) (* TODO: Remove this when https://github.com/roburio/http-lwt-client/issues/7 is fixed *)
-    ~meth
-    ~headers
-    url
-    (* TODO: This won't work once we handle things that aren't just short and simple JSON *)
-    (fun _ acc body -> Lwt.return (Some (Option.value ~default:"" acc ^ body)))
-    None
-  >|= function
-  | Ok ({Http_lwt_client.status = `OK; _}, Some body) -> Ok body
+  match
+    Httpcats.request
+      ~config:(`HTTP_1_1 H1.Config.default) (* TODO: Remove this when https://github.com/roburio/http-lwt-client/issues/7 is fixed *)
+      ~meth
+      ~headers
+      ~uri:url
+      (* TODO: This won't work once we handle things that aren't just short and simple JSON *)
+      ~f:(fun _ _ acc body -> match acc, body with
```

Comment:
I feel like the new labels on these two arguments are unnecessary too

Reply:
The `f` label is, indeed, not as important as all that, but the `uri` label is important because it indicates that the string passed is actually interpreted as an _uri_. The `string` type can easily be a catch-all type (just look at the OCaml codebase...) and being able to specify that the string expected by request is the uri is I think a good approach.